server/src/controllers/chatroom.controller.ts

```diff
@@ -1,33 +1,87 @@
-import { Request, Response, NextFunction } from "express";
-
-export const createChatRoom = (
-	req: Request,
-	res: Response,
-	next: NextFunction
-) => {
-	// TODO: 채팅방 생성
+import { Request, Response } from "express";
+import * as ChatroomService from "../db/context/chatrooms";
+
+// 호준님 작업 후, req.user.id 처리 필요.
+
+const handleError = (res: Response, error: unknown) => {
+	console.log(error);
+
+	if (error instanceof Error) {
+		return res.status(500).json({ message: error.message });
+	} else {
+		return res.status(500).json({ message: "An unknown error occurred" });
+	}
+};
+
+export const createChatRoom = async (req: Request, res: Response) => {
+	const { name } = req.body;
+	const userId = req.user.id;
+
+	try {
+		const chatroomId = await ChatroomService.createChatroom(name, userId);
+		return res.status(201).json({ chatroomId });
+	} catch (error) {
+		return handleError(res, error);
+	}
+};
+
+export const getChatrooms = async (req: Request, res: Response) => {
+	const userId = req.user.id;
+
+	try {
+		const chatrooms = await ChatroomService.getChatrooms(userId);
+		return res.status(200).json({ chatrooms });
+	} catch (error) {
+		return handleError(res, error);
+	}
 };
 
-export const readChatRoom = (
-	req: Request,
-	res: Response,
-	next: NextFunction
-) => {
-	// TODO: 채팅방 조회
+export const updateChatroom = async (req: Request, res: Response) => {
+	const { chatroomId } = req.params;
+	const numericChatroomId = parseInt(chatroomId, 10);
+	const { name } = req.body;
+
+	try {
+		await ChatroomService.updateChatroom(numericChatroomId, name);
+		res.status(200).json({ message: "채팅방이 수정되었습니다." });
```

Comment:
`nit` @jjhoooon 
크게 중요하지는 않지만, return 키워드를 명시적으로 사용해서 응답 이후 종료됨을 명확히 하는 것도 좋을 것 같습니다.
아래에도 같은 패턴이 반복되지만 코멘트는 여기만 작성하겠습니다.
```suggestion
		return res.status(200).json({ message: "채팅방이 수정되었습니다." });
```